api/src/test/java/org/openmrs/PersonTest.java

```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
+	    Assert.assertEquals(null, person.getAttribute("name3"));
+	}
+
+	/**
+	 * @see Person#getAttribute(PersonAttributeType)
+	 * @verifies null if all voided
+	 */
+	@Test
+	public void  getAttribute_shouldGetPersonAttributeBasedOnPersonAttributeTypeNullIfVoided() throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	 	PersonAttributeType type = new PersonAttributeType(new Integer(3));
+	 	type.setName("name3");
+	    Assert.assertEquals(null, person.getAttribute(type));
+	}
+
+	/**
+	 * @see Person#getAttribute(Integer)
+	 * @verifies get attribute based on Integer attributetypeid
+	 */
+	@Test
+	public void  getAttribute_shouldGetPersonAttributeBasedOnAttributeTypeId() throws Exception {
+		Person person = personHelper(false, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(new Integer(3), person.getAttribute(new Integer(3)).getAttributeType().getId());
+	}
+
+	/**
+	 * @see Person#getAttribute(Integer)
+	 * @verifies null if all voided, get attribute based on String attributetypeid
+	 */
+	@Test
+	public void  getAttribute_shouldGetPersonAttributeBasedOnAttributeTypeIdNullIfVoided() throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(null, person.getAttribute(new Integer(3)));
+	}
+
+	/**
+	 * @see Person#getAttributes(String)
+	 * @verifies get attributes based on String attributename
+	 */
+	@Test
+	public void  getAttributes_shouldGetPersonAttributesBasedOnAttributeName() throws Exception {
+		Person person = personHelper(false, 1, 2, 3, "name1", "name1", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(2, person.getAttributes("name1").size());
+	}
+
+	/**
+	 * @see Person#getAttributes(Integer)
+	 * @verifies get attributes based on Integer attributetypeid
+	 */
+	@Test
+	public void  getAttributes_shouldGetPersonAttributesBasedOnAttributeTypeId() throws Exception {
+		Person person = personHelper(false, 1, 1, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(2, person.getAttributes(new Integer(1)).size());
+	}
+
+/**
+	 * @see Person#getAttributes(Integer)
+	 * @verifies get attributes based on Integer attributetypeid, null if voided
+	 */
+	@Test
+	public void  getAttributes_shouldGetPersonAttributesBasedOnAttributeTypeIdNullIfVoided() throws Exception {
+		Person person = personHelper(true, 1, 1, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(0, person.getAttributes(new Integer(1)).size());
+	}
+
+private Person personHelper(boolean isVoid, int attributeType1, int attributeType2, int attributeType3, String attributeName1, String attributeName2, String attributeName3, String attributeValue1, String attributeValue2, String attributeValue3) {
```

Comment:
The formatting here looks ugly. Did you run mvn clean install before committing?